added info about an arabic script fix, fixed typo

semaeostomea opened the pull request:
related to #487 

As requested I added the info and code snippet for the arabic script fix to `docs/Text.md` and `docs/Unicode.md`

I also noticed that Hangul was mentioned as a script that's not supported, which isn't the case as you can see here:
![Screenshot 2022-08-16 175021](https://user-images.githubusercontent.com/100974908/184928878-d4ea0a22-c2e4-44e3-9d46-8ccfb0991020.png)
which is why I removed it, I added Kannada and Tamil as additional examples instead because they were mentioned in #474 and #365 

I also changed "Hindic" to "Indic" (note: Hindi is one of many Indian languages and uses Devanagari, "Hindic scripts" do not exist, I assume this was a typo)

- [x] A mention of the change is present in `CHANGELOG.md`
^ This is my first pull request, I'm not sure if something like this warrants a changelog entry

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/PyFPDF/fpdf2/blob/master/LICENSE).

Lucas-C commented on docs/Unicode.md:
```diff
@@ -54,6 +54,15 @@ From the error in the image above, Thai characters can be fixed using fonts from
 
 ![thai-font-working](https://raw.githubusercontent.com/PyFPDF/fpdf2/master/tutorial/thai-accent-working.png)
 
+*Scripts with automatic ligatures and contextual forms (eg. Indic scripts and Abjads like Devanagari, Hebrew, Arabic) currently have limited support. For the Arabic script there is a temporary solution (using two additional libraries) that works for most languages; only a few (rare) characters aren't supported. Using it on other scripts(eg. when the input is unknown or mixed scripts) does not aff ...
```
This section is already a bit long,
would you mind starting this section with a `### Arabic script workaround` header please?

semaeostomea replied:
done. Should I add it to `Text.md` too or is it fine like that?

Lucas-C replied:
It's fine like that 😊

Lucas-C reviewed (approved):
❤️ Thank you for this PR!

Would you mind please also adding a short mention of this workaround in `CHANGELOG.md`?

Something like this:
```markdown
## [2.5.7] - not released yet
### Added
- workaround by @semaeostomea to support the arabic script: [link to documentation](https://pyfpdf.github.io/fpdf2/Unicode.html#arabic-script-workaround)
```

Lucas-C commented:
@all-contributors please add @semaeostomea for documentation

gmischler commented:
Welcome @semaeostomea 
Good to have someone with practical experience in these matters chime in!
My own theoretical understanding is evidently incorrect at times...

The fact that arabic_reshaper can prepare text sufficiently for us to process directly indicates that there are seperate Unicode code points for each positional variant. Given that, your solution should really be integrated in fpdf2 directly. For the time being it is of course great to have it documented as a workaround.
Maybe you could add an indication of which names those packages have on PyPi? It looks to me like eg. `bidi` is listed as `python-bidi` there, which may be confusing to a newcomer.

Do you happen to know of any other scripts where all positional variants have seperate code points? If there are any, a final solution should treat them all this way.

I happen to be somewhat familiar with Mongolian, and I haven't found any positional code points for that (or Uighur, Manchu, etc.). In cases like this, we'd have to rely on substitutions offered by the font file. Can you confirm this?

I'm not very familiar with Hangul. Since those caracter are technically composites and require special input methods to type, I expected them to be handled like ligatures in Unicode as well. Your remark indicates that I was mistaken about this, and each combination is indeed a seperate code point. So much the better!

Sorry about the hindic/indic confusion, I tend to fall for that once in a while... :wink:

Lucas-C commented on CHANGELOG.md:
```diff
@@ -394,3 +394,7 @@ prevented strings passed first to the text-rendering methods to be displayed.
 * unit tests now use the standard `unittest` lib
 * massive code cleanup using `flake8`
+
+## [2.5.7] - not released yet
```
This should be at the top of the `CHANGELOG.md` file :)
You may have to rebase your branch because I just merged PR #492 that added an entry to this file too

semaeostomea commented:
> Do you happen to know of any other scripts where all positional variants have seperate code points? If there are any, a final solution should treat them all this way.
Not that I know of. Most of them lack a proper Unicode in the first place, which is also a problem with Mongolian as you probably know

> I happen to be somewhat familiar with Mongolian, and I haven't found any positional code points for that (or Uighur, Manchu, etc.). In cases like this, we'd have to rely on substitutions offered by the font file. Can you confirm this?
If you're referring to the old Uyghur script, yes, and afaik the Unicode doesn't even cover it completely but I'm not really familiar with it.
As for the Uyghur adapted Arabic script that is in use nowadays, this is also covered by my solution (as are all other languages that use the Arabic script. Some languages that use it have rare special characters that I noticed aren't covered, but those are exceptions and the majority of the script is still displayed correctly)

I also forgot to mention that Hebrew doesn't have contextual forms. It's displayed wrong because it's RTL like Arabic. But putting it through the same code fixes that too. The only problem is that the diacritics/marks are displaced (not everyone uses them though). Depending on which font is used they're more or less displaced
![Screenshot 2022-08-17 093451](https://user-images.githubusercontent.com/100974908/185070869-10fb89d8-2219-459c-924c-a895db441407.png)
כַּף סוֹפִית :correct
This is a common problem with lots of software and a number of scripts with diacritics that have separate code points though. I mostly notice this with Diné bizaad and when someone uses a different input method for Czech that separates the letter and diacritics, they end up all over the place.
I just tested it and it's the same problem with Diné bizaad:
![Screenshot 2022-08-17 102804](https://user-images.githubusercontent.com/100974908/185072112-30b55989-6697-47f7-b38e-3badd7e22804.png)
łį́į́ʼ in the middle

semaeostomea commented:
I just changed "Arabic script workaround" to "Right to Left & Arabic script workaround" because it generally also fixes the out-of-order problem of RTL scripts with bidi.
I also added two points to mention the RTL and diacritic issue in `Text.md` because it's not covered by the other ones and put the header in there after all too because it got a bit long too.
I hope it's alright like that.

Lucas-C commented:
This PR looks great to me!

@gmischler: would you add anything or are your OK to merge it?

gmischler commented:
> Not that I know of. Most of them lack a proper Unicode in the first place, which is also a problem with Mongolian as you probably know

As far as I understand, Mongolian (et. al) Unicode is complete for the isolated forms.
There should be a table in the font listing the other shapes by context.

> The only problem is that the diacritics/marks are displaced (not everyone uses them though).
> Depending on which font is used they're more or less displaced
> ![Screenshot 2022-08-17 093451](https://user-images.githubusercontent.com/100974908/185070869-10fb89d8-2219-459c-924c-a895db441407.png)
>  כַּף סוֹפִית :correct 
> This is a common problem with lots of software and a number of scripts with diacritics that have separate code points though.
> 
 > I mostly notice this with Diné bizaad and when someone uses a different input method for Czech that separates the letter and diacritics, they end up all over the place. I just tested it and it's the same problem with Diné bizaad: ![Screenshot 2022-08-17 102804](https://user-images.githubusercontent.com/100974908/185072112-30b55989-6697-47f7-b38e-3badd7e22804.png)
> łį́į́ʼ in the middle

Shouldn't some type of Unicode normalization fix those diacritics?
`unicodedata.normalize('NFC', 'łį́į́ʼ')`
(or maybe 'NFKC', didn't do any tests right now).

Come to think of it, I'll have to test later if this would actually solve #459.

gmischler commented:
> @gmischler: would you add anything or are your OK to merge it?

No obstacles that I can see.

semaeostomea commented:
> As far as I understand, Mongolian (et. al) Unicode is complete for the isolated forms. There should be a table in the font listing the other shapes by context.

The Mongolian Unicode is complete, but there are some problems with it. I don't understand the underlying mechanism enough, I just know that it can result in the wrong form being displayed sometimes.
If you don't know it yet, [this](https://www.w3.org/TR/mong-gap/) is a good read, it mentions some of the problems with the Unicode

> Shouldn't some type of Unicode normalization fix those diacritics? `unicodedata.normalize('NFC', 'łį́į́ʼ')` (or maybe 'NFKC', didn't do any tests right now).

This can fix come diacritics but unfortunately not all. I can imagine it would fix Czech diacritics because those exist as a separate code point when combined with the letter (if that's how it works). That's not the case for some diacritics and tone marks in Diné bizaad because they're not used in any other language combined like that
![Screenshot 2022-08-17 114453](https://user-images.githubusercontent.com/100974908/185088339-288e88cd-ede8-45a8-8c0d-c2593a1ac51b.png)
